frontend/app/teams/layout.js

```diff
@@ -1,18 +1,52 @@
 'use client';
-import { useState, createContext } from 'react';
+import { useEffect, useState } from 'react';
+import { auth } from '@firebase/config';
+import { onAuthStateChanged } from 'firebase/auth';
+import Cookies from 'js-cookie';
+import { getUser, createUser } from '@airtable/utils';
 import Nav from '@components/Nav/ParentNav/ParentNav';
 import teamsStyle from './Teams.module.css';
-
-export const TeamsContext = createContext(null);
+import { UserContext } from '@user-context';
+import { useRouter } from 'next/navigation';
+import Loader from '@components/Loader/Loader';
 
 export default function TeamLayout({ children }) {
-  const [teamList, setTeamList] = useState([]);
+  const router = useRouter();
+  const [user, setUser] = useState([]);
+  const value = { user, setUser };
+
+  useEffect(() => {
+    const handleLoginOnPageLoad = onAuthStateChanged(auth, async (firebaseUser) => {
+      if (!firebaseUser) {
+        Cookies.remove('uid');
+        router.push('/');
+      }
+
+      if (firebaseUser) {
+        const { uid, email } = firebaseUser;
+        const doesUserExist = await getUser(uid);
+        const hasAirtableRecord = doesUserExist.users.length !== 0;
+        let userData = doesUserExist.users;
+
+        if (!hasAirtableRecord) {
+          const createdUser = await createUser(uid, email);
+          userData = createdUser.users;
+        }
+        setUser(userData);
+      }
+    });
+
+    return () => handleLoginOnPageLoad();
+  }, []);
+
+  if (!user.length) return <Loader />;
+
   return (
-    <div className={teamsStyle.grid}>
-      <TeamsContext.Provider value={{ teamList, setTeamList }}>
-        <Nav teams={teamList} />
+    <UserContext.Provider value={value}>
```

Comment:
I would also refactor the context functionality into the useContext file. So that whenever you need to add to it or change it, there is a clear file for it. I was looking in that file originally for what the user context returns, it is not intuitive that it would be located in the Layout component.